x-pack/plugins/security_solution/server/endpoint/services/actions/utils.ts

```diff
@@ -130,13 +133,21 @@ export const getActionCompletionInfo = (
   // If completed, then get the completed at date and determine if action was successful or not
   if (completedInfo.isCompleted) {
     const responseErrors: ActionCompletionInfo['errors'] = [];
-
-    for (const normalizedAgentResponse of Object.values(responsesByAgentId)) {
+    completedInfo.outputs = {};
+    for (const agentKey of Object.keys(responsesByAgentId)) {
```

Comment:
Changed this as we need `agentId` below for `outputs`

Reply:
sounds good.  I would suggest renaming `agentKey` to `agentId` just to clarify what the value is.

FYI - could also have used `Object.entries()` and destruct that - like: 

```typescript
for (const [agentId, normalizedAgentResponse] of Object.entries(responsesByAgentId)) {
```